src/main/java/net/openhft/chronicle/queue/impl/single/TableStoreWriteLock.java

```diff
@@ -69,40 +69,57 @@ public void lock() {
         try {
             currentLockValue = lock.getVolatileValue();
             while (!lock.compareAndSwapValue(UNLOCKED, PID)) {
-                if (Thread.currentThread().isInterrupted())
-                    throw new InterruptedRuntimeException("Interrupted for the lock file:" + path);
-                tlPauser.pause(timeout, TimeUnit.MILLISECONDS);
-                currentLockValue = lock.getVolatileValue();
+                currentLockValue = lockGetCurrentLockValue(tlPauser);
             }
 
-            //noinspection ConstantConditions,AssertWithSideEffects
-            assert (lockedByThread = Thread.currentThread()) != null
-                    && (lockedHere = new StackTrace()) != null;
+            lockAssertPostConditions();
 
             // success
         } catch (TimeoutException e) {
-            final String lockedBy = getLockedBy(currentLockValue);
-            final String warningMsg = "Couldn't acquire write lock " +
-                    "after " + timeout + " ms " +
-                    "for the lock file:" + path + ". " +
-                    "Lock was held by " + lockedBy;
-            if (forceUnlockOnTimeoutWhen == UnlockMode.NEVER)
-                throw new UnrecoverableTimeoutException(new IllegalStateException(warningMsg + UNLOCK_MAIN_MSG));
-            else if (forceUnlockOnTimeoutWhen == UnlockMode.LOCKING_PROCESS_DEAD) {
-                if (forceUnlockIfProcessIsDead())
-                    lock();
-                else
-                    throw new UnrecoverableTimeoutException(new IllegalStateException(warningMsg + UNLOCK_MAIN_MSG));
-            } else {
-                warn().on(getClass(), warningMsg + UNLOCKING_FORCIBLY_MSG);
-                forceUnlock(currentLockValue);
-                lock();
-            }
+            lockHandleTimeoutException(currentLockValue);
         } finally {
             tlPauser.reset();
         }
     }
 
+    private long lockGetCurrentLockValue(TimingPauser tlPauser) throws TimeoutException {
+        if (Thread.currentThread().isInterrupted())
+            throw new InterruptedRuntimeException("Interrupted for the lock file:" + path);
+        tlPauser.pause(timeout, TimeUnit.MILLISECONDS);
+        return lock.getVolatileValue();
+    }
+
+    private void lockAssertPostConditions() {
+        //noinspection ConstantConditions,AssertWithSideEffects
+        assert (lockedByThread = Thread.currentThread()) != null
```

Comment:
consider using `assert AssertUtil.SKIP_ASSERTIONS || ` which is a further watered-down flavour of assertion that we use

Reply:
Thanks for the tip Jerry - looks like this assertion lib is not on the classpath of Queue at the moment - let me add it in a separate PR.